### ISSUE TITLE
voxelmorph2d results

### ISSUE THREAD

**Issue by tianyuntong1020** [Hsankesara/VoxelMorph-PyTorch#3]
Hi，I use your code about voxelmorph 2d,but results are not good.Could you help me check it .I don't know how to solve this problem. Results like this:
![reg22](https://user-images.githubusercontent.com/52816447/63987588-5f921a00-cb0b-11e9-81c0-f7997e26bdc3.png)
![reg32](https://user-images.githubusercontent.com/52816447/63987589-628d0a80-cb0b-11e9-84ba-f18e889dcfe1.png)

**Comment by Hsankesara** [comment]
Het @tianyuntong1020, I may have committed the wrong code by mistake. That is why the problem is occurring. I'll update it as soon as possible. Apologies.

**Comment by tianyuntong1020** [comment]
That's all right.Looking forward to your quick solution.

**Comment by tianyuntong1020** [comment]
Is there something wrong with the networks module ?such as Unet, SpatialTransformation structure,because I think results are not good perhaps depend on networks,but I don't found error about it.

**Comment by Hsankesara** [comment]
I think the loss function is erroneous. In smoothing loss, I've used the registered image for the loss but in the paper, they have used deformation vector field for that. It is a small modification in the code and I'll change it soon. I'm trying to reproduce the results as same as Tensorflow code and will update you when it is done.

**Comment by tianyuntong1020** [comment]
Thanks for your reply. I got it, but I mean that loss function doesn't impact registration output image, it just evaluate performance .The results occur black image, I wonder if the reproduction of network structure is correct.I use my own dataset with coarse registration to try your code,results don't previous phenomenon. So,I want to refer your opinion. Thank you a lot.

**Comment by HP192080042** [comment]
Hi, author, I use your code about voxelmorph 2d,but results are not good？
![Snipaste_2019-11-21_15-17-31](https://user-images.githubusercontent.com/53810832/69315855-85463f80-0c72-11ea-8ea1-ffad762a078d.png)
Looking forward to your update！

**Comment by HP192080042** [comment]
@Hsankesara

**Comment by Hsankesara** [comment]
Sorry, @HP192080042  for the results. I'll make the updates in the code as soon as I can.

**Comment by davidz123** [comment]
@Hsankesara I've got the same result above,  looking forward to your code updated.

**Comment by B1GGUN** [comment]
> Het @tianyuntong1020, I may have committed the wrong code by mistake. That is why the problem is occurring. I'll update it as soon as possible. Apologies.
Hello, is the code still being updated?

**Comment by Paluck1Arora2** [comment]
> 

Hi

> Hi，I use your code about voxelmorph 2d,but results are not good.Could you help me check it .I don't know how to solve this problem. Results like this: ![reg22](https://user-images.githubusercontent.com/52816447/63987588-5f921a00-cb0b-11e9-81c0-f7997e26bdc3.png) ![reg32](https://user-images.githubusercontent.com/52816447/63987589-628d0a80-cb0b-11e9-84ba-f18e889dcfe1.png)

Hi, is the code being updated??

**Comment by domadaaaa** [comment]
4 years passed...

**Comment by Hsankesara** [comment]
Hi @domadaaaa, fair enough. Unfortunately, I don't have time now to work on it. I am happy if someone else wants to take this on.

**Comment by domadaaaa** [comment]
> Hi @domadaaaa, fair enough. Unfortunately, I don't have time now to work on it. I am happy if someone else wants to take this on.

alright, hope everything goes well with your work!

**Comment by BaIGaO** [comment]
Same issue. This was caused by an incorrect input of the smoothing-loss. The input of smoothing-loss in the original paper was the deformation field, but this code got it wrong.I have fixed this issue in my repo, it's works well. https://github.com/BaIGaO/VoxelMorph-PyTorch
